api/src/main/java/org/openmrs/module/queue/model/QueueEntry.java

```diff
@@ -80,6 +80,11 @@ public class QueueEntry extends BaseChangeableOpenmrsData {
 	@JoinColumn(name = "provider_waiting_for", referencedColumnName = "provider_id")
 	private Provider providerWaitingFor;
 	
+	//The Location the patient is coming for, if any.
+	@ManyToOne
```

Comment:
Why `@ManyToOne`?

Reply:
multiple instances of location_coming_from can be associated with a single queue

Reply:
It can be confusing whether to use one/many to one relationship. @corneliouzbett - I wonder what this will look like for those who are sent to the lab but has to go through billing. Based on the current design, what are your thoughts?

Reply:
`@ManyToOne` is okay for this case.
However, I'm curious if the next queue is determined based on the client's origin.

Reply:
Based on the current design, the clinical can send the client to billing and then from billing to the Lab (Clinical consultation -> Billing -> Lab -> Clinical Consultation (again)).
I suggest enabling clinical officers to control client movement. Once one stage is done, the client can be automatically placed in the next queue with a designated priority.

Reply:
> `@ManyToOne` is okay for this case. However, I'm curious if the next queue is determined based on the client's origin.

This is dependent on the clinic workflow. For now, the providers chose the queue that they want to move a patient to, its not done automatically yet

Reply:
I concur with @corneliouzbett . With the current design we have, the patient will only have one active queue at a time. One can not belong to two different active queues. Once a patient is moved from one queue to another, the previous queue is ended.

Reply:
Done